judge/forms.py

```diff
@@ -69,13 +69,19 @@ def __init__(self, *args, **kwargs):
 
 class ProblemSubmitForm(ModelForm):
     source = CharField(max_length=65536, widget=AceWidget(theme='twilight', no_ace_media=True))
+    judge = ChoiceField(choices=(), widget=forms.HiddenInput(), required=False)
 
     def __init__(self, *args, **kwargs):
+        judge_queryset = kwargs.pop('judge_queryset', ())
         super(ProblemSubmitForm, self).__init__(*args, **kwargs)
         self.fields['language'].empty_label = None
         self.fields['language'].label_from_instance = attrgetter('display_name')
         self.fields['language'].queryset = Language.objects.filter(judges__online=True).distinct()
 
+        if judge_queryset:
+            self.fields['judge'].widget = Select2Widget(attrs={'style': 'width:150px', 'data-placeholder': _('any')})
```

Comment:
(I know `any` was what the mockup had :)

```suggestion
            self.fields['judge'].widget = Select2Widget(attrs={'style': 'width:150px', 'data-placeholder': _('Any judge')})
```